src/ProjectConfiguration.ts

```diff
@@ -254,6 +261,40 @@ export function normalizeProjectConfig(
     );
   }
 
+  /**
+   * FbOS 3.0 has a max size limit on the app manifest of 1K
+   * so if a developer specifies language + locale variants (eg en and en-US)
+   * for all languages they'll quickly exceed this. This code merges the language
+   * tag into a locale tag unless one is present.
+   */
+  const culledLocales: string[] = [];
+  mergedConfig.i18n = lodash.mapKeys(mergedConfig.i18n, (_, locale) => {
+    const localeMapping: { [locale: string]: string | undefined } = {
+      en: 'en-US',
+      de: 'de-DE',
+      it: 'it-IT',
+      es: 'es-ES',
+      fr: 'fr-FR',
+      ja: 'ja-JP',
+      ko: 'ko-KR',
+      nl: 'nl-NL',
+      sv: 'sv-SE',
+    };
+    const mappedLocale = localeMapping[locale];
+
+    if (mappedLocale === undefined) {
+      return normalizeLanguageTag(locale);
+    }
+
+    if (mergedConfig.i18n[mappedLocale] !== undefined) {
+      culledLocales.push(locale);
+      return locale;
+    }
+
+    return mappedLocale;
+  });
+  mergedConfig.i18n = lodash.omit(mergedConfig.i18n, culledLocales);
```

Comment:
why do we have to ommit? doesn't lodash.mapKeys remap the `locale` key to the proper mappedLocale?

Reply:
This handles the case where both say `en-US` and `en` appear. `en` gets mapped to `en-US`, but if we already have `en-US` we don't want to overwrite it. I've re-written this without lodash to make it clearer.